pkg/api/customization/globalrole/validator.go

```diff
@@ -5,18 +5,19 @@ import (
 
 	"github.com/rancher/norman/types"
 	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 type Wrapper struct {
-	GlobalRoleLister v3.GlobalRoleLister
+	GlobalRoleClient v3.GlobalRoleInterface
 }
 
 func (w Wrapper) Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
 	if request.Method != http.MethodPut {
 		return nil
 	}
 
-	gr, err := w.GlobalRoleLister.Get("", request.ID)
+	gr, err := w.GlobalRoleClient.Get(request.ID, v1.GetOptions{})
```

Comment:
do you think it'd make sense to first check Lister and use client only if errors isNotFound?

Reply:
This should go back to using the lister. The test is doing something atypical for how a grb is going to be used.